### DISCUSSION TITLE
Connection reset by peer when using custom host and port

### DISCUSSION THREAD

**Discussion by Mubelotix** [immich-app/immich#10329] [Q&A] [answered]
### The bug

Immich resets all connections on the given port with the env file given below. Commenting IMMICH_PORT and IMMICH_HOST resolves the issue (but I can't chose the port)

### The OS that Immich Server is running on

Debian

### Version of Immich Server

v1.106.4

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Downloaded from github, not altered
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

UPLOAD_LOCATION=./library
DB_DATA_LOCATION=./postgres
IMMICH_VERSION=v1.106.4 # Change this when updating
DB_PASSWORD=redacted
TZ="Europe/Paris"
IMMICH_HOST=127.0.0.1
IMMICH_PORT=2283
UPLOAD_LOCATION=/data/immich

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich
```

### Reproduction steps

```bash
1. Run the docker compose
2. Attempt to make requests
```

**Comment by mmomjian** [comment, marked as answer]
Please provide logs.

what is `{{immich_inner_port}}` ? This seems like a pretty complex deployment.

Why are you listening on 127.0.0.1? As far as I know, within Docker that means it won't be able to respond to any requests from the host (unless you also use network_mode: host, but you said you didn't edit the docker-compose...)

Why do you have UPLOAD_LOCATION specified twice in the .env?

**Reply by Mubelotix** [reply]
> what is {{immich_inner_port}} ? This seems like a pretty complex deployment.

It's replaced with `2283` by ansible, I edited the original post with the actual env file.

> Why are you listening on 127.0.0.1? As far as I know, within Docker that means it won't be able to respond to any requests from the host (unless you also use network_mode: host, but you said you didn't edit the docker-compose...)

That might be the reason then, thank you

> Why do you have UPLOAD_LOCATION specified twice in the .env?

That was by mistake but fortunately the latter was the correct one

**Comment by bo0tzz** [comment]
You mention the compose file is not altered - if you're gonna change the port then you certainly need to alter the compose file as well.

**Comment by say4n** [comment]
See this for diagnosis & solution: https://github.com/immich-app/immich/discussions/13568
